test/distributed-ranges/shp/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+# SPDX-FileCopyrightText: Intel Corporation
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+if(${CMAKE_VERSION} VERSION_LESS "3.26")
+    message(WARNING "disabled distributed ranges tests, use cmake>=3.26")
+    return()
+else ()
+    message(STATUS "suitable for distributed ranges cmake version ${CMAKE_VERSION} found")
+endif()
+
+find_package(fmt QUIET)
+if(fmt_FOUND)
+    if(fmt_VERSION VERSION_GREATER_EQUAL 8.1 AND fmt_VERSION VERSION_LESS 10)
+        message(STATUS "suitable for distributed ranges libfmt version ${fmt_VERSION} found")
+    else()
+        message(WARNING "libfmt version ${fmt_VERSION} is not >= 8.1 and < 10, skipping distributed ranges tests")
+        return()
+
+    endif()
+else()
+    message(WARNING "libfmt not found, install libfmt-dev 8.x or 9.x to enable distributed ranges tests")
+    return()
+endif()
+
+find_package(GTest)
+if (NOT GTest_FOUND)
+  message(WARNING "disabled distributed ranges tests, install libgtest-dev or other package with GTest")
+  return()
+endif ()
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+add_executable(
+  shp-tests
+  shp-tests.cpp 
+  ../common/all.cpp 
```

Comment:
I notice `[inclusive/exclusive]_scan.cpp` and `sycl_utils.cpp` are missing from this list.  Is that intentional?

Similarly, there are some tests missing from this directory as well.
In the main tests CMakeLists for oneDPL, we let the file structure dictate the test lists, so no extra step is required to add a test to the suite, which is nice because it avoids errors of omission, silently skipping tests we think are there.

For kernel templates, we have many combinations of params, so we generate the test list (kernel templates), but that is very much more intentional.

This isn't a requirement to switch, but may be worth evaluating.

Reply:
Thank you for finding this. I've added all missing files and reported a task to refactor our CMakeLists in the way you suggested.